src/layers/transform/evaluation.cpp

```diff
@@ -37,30 +37,31 @@ namespace {
 /** CPU implementation of evaluation layer forward prop. */
 void fp_cpu(lbann_comm& comm,
             const AbsDistMat& input,
-            DataType& value,
-            Al::request& req) {
+            DataType& value) {
   const auto& local_input = input.LockedMatrix();
   const auto& local_height = local_input.Height();
   const auto& local_width = local_input.Width();
   const auto& mini_batch_size = input.Width();
-  value = DataType(0);
+  value = 0;
 #pragma omp parallel for reduction(+:value) collapse(2)
   for (El::Int col = 0; col < local_width; ++col) {
     for (El::Int row = 0; row < local_height; ++row) {
       value += local_input(row, col);
     }
   }
   value = value / mini_batch_size;
-  comm.nb_allreduce(&value, 1, input.DistComm(), req);
+  comm.allreduce(&value, 1, input.DistComm());
```

Comment:
Why does this become blocking for the CPU path?  Shouldn’t it remain independent of the GPU path?

Reply:
This was mostly for simplicity. I can restore the non-blocking CPU implementation if need be.